lib/Doctrine/ODM/PHPCR/DocumentManager.php

```diff
@@ -925,4 +925,16 @@ public function getNodeForDocument($document)
 
         return $this->session->getNode($path);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDocumentId($document)
+    {
+        try {
+            return $this->unitOfWork->getDocumentId($document);
+        } catch (PHPCRException $e) {
```

Comment:
I think different. When doing `find()` we also eat several exceptions. When having a new document which isn't managed yet,  a null as ID sounds valid. Just the "not managed due to problems" aspects would confuse with null as a return value.